### PR TITLE
feat: add Subscription

### PR DESCRIPTION
**Pull request by popomore**
##### Checklist

- [x] `npm test` passes
- [ ] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message follows commit guidelines

##### Description of change

Ref https://github.com/eggjs/egg/issues/1468

### CODE REVIEW

**Comment by atian25** [comment]
@popomore 这个也要发掉了，不然 egg-schedule 的新用户按文档写会报错

**Comment by atian25** [comment]
ping

**Comment by popomore** [comment]
@atian25 看看

File: docs/source/en/basics/objects.md [hunk 1]
```diff
@@ -270,6 +270,22 @@ We can get it via `ctx.coreLogger`, the difference between the Context Logger is
...
+
+Subscription is the model for subscribing, including consumer in message broker or schedule.
+
+The base class of Subscription is exported by egg.
+
+```js
+const Subscription = require('egg').Subscription;
+class Schedule extends Subscription {
+  // This method should be implemented
+  * subscribe() {}
```
Comment (atian25) [inline docs/source/en/basics/objects.md]: 这里可以是 generator / promise / async 吧?

Comment (popomore) [reply docs/source/en/basics/objects.md]: 是的

Comment (atian25) [reply docs/source/en/basics/objects.md]: 嗯，所以文档要注明下？

Comment (atian25) [reply docs/source/en/basics/objects.md]: `subscribe` could be `generator / async function`

File: docs/source/zh-cn/basics/objects.md [hunk 2]
```diff
@@ -270,6 +270,22 @@ module.exports = {
 
 我们可以在 Controller 和 Service 实例上通过 `this.logger` 获取到它们，它们本质上就是一个 Context Logger，不过在打印日志的时候还会额外的加上文件路径，方便定位日志的打印位置。
 
+## Subscription
+
+Subscription 是一种订阅模型，消息中间件的消费者或调度任务都属于这种模式。
```
Comment (atian25) [inline docs/source/zh-cn/basics/objects.md]: `订阅模型是一种比较常见的开发模式，譬如消息中间件的消费者或调度任务。因此我们提供了 Subscription 基类来规范化这个模式`

**Pull request merged**